ports/stm32/machine_adc.c

```diff
@@ -42,10 +42,18 @@
 #define ADCx_COMMON __LL_ADC_COMMON_INSTANCE(0)
 #endif
 
+#if defined(STM32G4)
+#define ADC_CHANNEL_TEMPSENSOR ADC_CHANNEL_TEMPSENSOR_ADC1
+#define ADC_CCR_TSEN ADC_CCR_VSENSESEL
+#define ADC_CCR_VBATEN ADC_CCR_VBATSEL
+#endif
```

Comment:
Instead of defining these macros here to match other MCU's, I think it would be better to instead modify the code in this file to have different code for the STM32G4 (as far as I can tell, it's only one location in `adc_config_channel` and there is already special code there for G4).

Also, the G4 is already handled separately for `ADC_CHANNEL_TEMPSENSOR_ADC1` in the definition of `MP_QSTR_CORE_TEMP`.

Reply:
Thank you for reviewing.
Yes, it's better instead of defining macros. Now I fixed it (9001799d).